src/settings.rs

```diff
@@ -1428,4 +1579,50 @@ mod tests {
 
         assert_eq!(default, res);
     }
+
+    #[meilisearch_test]
+    async fn test_get_typo_tolerance(index: Index) {
+        let typo_tolerance = TypoToleranceSettings::default();
+
+        let res = index.get_typo_tolerance().await.unwrap();
+
+        assert_eq!(typo_tolerance, res);
+    }
+
+    #[meilisearch_test]
+    async fn test_set_typo_tolerance(client: Client, index: Index) {
+        let typo_tolerance = TypoToleranceSettings {
+            enabled: Some(true),
+            disable_on_attributes: Some(vec!["title".to_string()]),
+            disable_on_words: Some(vec![]),
+            min_word_size_for_typos: Some(MinWordSizeForTypos::default()),
+        };
+
+        let task_info = index.set_typo_tolerance(&typo_tolerance).await.unwrap();
+        client.wait_for_task(task_info, None, None).await.unwrap();
+
+        let res = index.get_typo_tolerance().await.unwrap();
+
+        assert_eq!(typo_tolerance, res);
+    }
+
+    #[meilisearch_test]
```

Comment:
Can you add a test where you only send a partial object ? 
for example:

- update only `enabled`
- make the request, check if correct

```rust
 let typo_tolerance = TypoToleranceSettings {
    enabled: Some(false),
};
```

in the same test: 
- update only `disabled_on_attributes`
- make the request, check if both `enabled` and `disabled_on_attributes` still have the correct changes.

Reply:
I have to initialize all the fields of a struct. I cannot have a partial struct. Is it possible?

Reply:
`

      missing structure fields:
      - disable_on_attributes
      -  disable_on_words
     - min_word_size_for_typos

`

Reply:
Oh yeah... I mixed up the languages. My bad. I guess there that's the reason for adding the builder pattern... If you don't mind can you add it? With all the default values set to None?

Very sorry about the mixup. A lot of context switches between languages😅

Reply:
If you look at this sample code https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d78d2a9f68131c270eadffb4df44d4a0; the changes I had made allowed for partial objects to be sent. The standard output shows - 



      Ok("{\"minWordSizeForTypos\":{\"oneTypo\":0},\"age\":10}")

Reply:
> If you look at this sample code https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d78d2a9f68131c270eadffb4df44d4a0; the changes I had made allowed for partial objects to be sent. The standard output shows -
> 
> ```
>   Ok("{\"minWordSizeForTypos\":{\"oneTypo\":0},\"age\":10}")
> ```

@bidoubiwa Please see the above sample code shared on playground.